src/nb_func.cpp

```diff
@@ -277,13 +275,13 @@ PyObject *nb_func_new(const void *in_) noexcept {
     check(func, "nb::detail::nb_func_new(\"%s\"): alloc. failed (1).",
           name_cstr);
 
-    func->max_nargs_pos = f->nargs;
-    func->complex_call = has_args || has_var_args || has_var_kwargs || has_keep_alive;
+    func->max_nargs = f->nargs;
+    func->complex_call = f->nargs_pos < f->nargs || has_args || has_keep_alive;
```

Comment:
Is there any particular reason that we need a less than comparison, instead of `!=`? I'm assuming `nargs_pos` can never actually be larger than `nargs`, correct? No need to change anything here, I am just wondering.

Reply:
Either would work; nargs_pos is always <= nargs, and always == for "simple" calls with no kwonly/varargs/varkwds.